rest/route/host_agent.go

```diff
@@ -525,6 +525,26 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 				"host_id":   currentHost.Id,
 			}))
 
+			if nextTask.IsStuckTask() {
+				if err := model.EndAndResetSystemFailedTask(ctx, env.Settings(), nextTask, evergreen.TaskDescriptionStranded); err != nil {
+					return nil, false, errors.Wrap(err, "ending and resetting system failed task")
+				}
+				// The agent requesting the same task over and over again is a sign that the agent may have hit a panic
+				// and is caught up in an endless loop. This doesn't shut down the host because when starting over again
+				// in that case it will likely cause the same panic again. This resets the task and logs so that the task
+				// doesn't have to  be manually unstuck and an alert can be set up that alerts engineers who can manually
+				// intervene and fix the agent -- likely via a revert.
+				msg := fmt.Sprintf("The agent has re-requested the same task '%d' times.", evergreen.MaxTaskDispatchAttempts)
+				msg += " It's possible that the agent hit a panic and is in a bad state."
+
+				grip.Error(message.Fields{
+					"message":   msg,
+					"distro_id": d.Id,
+					"task_id":   nextTask.Id,
+					"host_id":   currentHost.Id,
```

Comment:
Can we include the agent's revision in this message so we can determine if it's a bad agent deploy?